### PR TITLE
Spark: Add a test for MERGE modifying a null struct

### PR DESCRIPTION
**Pull request by aokolnychyi**
This PR adds a test for modifying a null struct in MERGE.

### CODE REVIEW

**Comment by aokolnychyi** [comment]
cc @dilipbiswal @rdblue

File: spark/src/test/java/org/apache/iceberg/spark/SparkTestBase.java [hunk 1]
```diff
@@ -97,11 +97,24 @@ protected long waitUntilAfter(long timestampMillis) {
...
 
-    return rows.stream()
-        .map(row -> IntStream.range(0, row.size())
-            .mapToObj(pos -> row.isNullAt(pos) ? null : row.get(pos))
-            .toArray(Object[]::new)
-        ).collect(Collectors.toList());
+    return rows.stream().map(this::toJava).collect(Collectors.toList());
+  }
+
+  private Object[] toJava(Row row) {
```
Comment (aokolnychyi) [inline spark/src/test/java/org/apache/iceberg/spark/SparkTestBase.java:103]: I had to rewrite it to support nested structs.

File: spark3-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestMerge.java [hunk 2]
```diff
@@ -49,9 +49,23 @@ public void removeTables() {
   }
 
   // TODO: tests for reordering when operations succeed (both insert and update actions)
-  // TODO: tests for modifying fields in a null struct
   // TODO: tests for subqueries in conditions
 
+  @Test
+  public void testMergeModifiesNullStruct() {
```
Comment (aokolnychyi) [inline spark3-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestMerge.java:55]: The behavior matches what I tested for UPDATE in Postgres.

**Review by dilipbiswal — APPROVED** [review]

**Comment by dilipbiswal** [comment]
LGTM . thanks for adding this test @aokolnychyi

**Review by rdblue — APPROVED** [review]

**Comment by rdblue** [comment]
Thanks, @aokolnychyi!
